src/openrct2/rct2/S6Importer.cpp

```diff
@@ -291,17 +290,30 @@ namespace OpenRCT2::RCT2
                 }
             }
 
-            // Localise the park name and description
-            StringId localisedStringIds[3];
-            if (LanguageGetLocalisedScenarioStrings(dst->Name, localisedStringIds))
+            // dst->name will be translated later so keep the untranslated name here
+            String::Set(dst->InternalName, sizeof(dst->InternalName), dst->Name);
+            String::Set(dst->Details, sizeof(dst->Details), _s6.Info.Details);
+
+            if (desc.textObjectId != nullptr)
             {
-                if (localisedStringIds[0] != STR_NONE)
-                {
-                    String::Set(dst->Name, sizeof(dst->Name), LanguageGetString(localisedStringIds[0]));
-                }
-                if (localisedStringIds[2] != STR_NONE)
+                auto& objManager = GetContext()->GetObjectManager();
+
+                // Ensure only one thread talks to the object manager at a time
+                std::lock_guard lock(mtx);
```

Comment:
Does this not require unlocking?

Reply:
a lock_guard auto unlocks on destruction of the variable.